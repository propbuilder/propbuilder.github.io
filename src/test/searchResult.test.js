import '@testing-library/jest-dom/extend-expect';
import SearchResult from "../components/searchResult";
import {
    cleanup,
    render
  } from '@testing-library/react';

describe('Search result test', () => {
    afterEach(cleanup);
    test('1. search result component should render properly', () => {
        const mockStockDetails = {
            Name: 'International Business Machines',
            Symbol: 'IBM',
            Description: 'Product org.',
            AnalystTargetPrice: '2.23',
            Beta: '11.1',
            Industry: 'Computers',
            PERatio: '12.2',
            MarketCapitalization: '123345'
        };
        const { container, getByText } = render(
            <SearchResult 
                stockDetails={mockStockDetails}
            />
        );
        expect(container).toMatchSnapshot();
        expect(getByText('Computers')).toBeInTheDocument();
        expect(getByText('Product org.')).toBeInTheDocument();
    });
});