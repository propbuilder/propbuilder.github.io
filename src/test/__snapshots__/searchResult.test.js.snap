// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search result test 1. search result component should render properly 1`] = `
<div>
  <div
    class="search-result"
    data-testid="searchResult"
  >
    <div
      class="detail"
    >
      1. Name and Symbol of the company
      <div
        class="detail-content"
      >
        International Business Machines
         
        IBM
      </div>
    </div>
    <div
      class="detail"
    >
      2. Description
      <div
        class="detail-content"
      >
        Product org.
      </div>
    </div>
    <div
      class="detail"
    >
      3. Current Price
      <div
        class="detail-content"
      >
        2.23
      </div>
    </div>
    <div
      class="detail"
    >
      4. Change its traded on
      <div
        class="detail-content"
      >
        11.1
      </div>
    </div>
    <div
      class="detail"
    >
      5. Industry
      <div
        class="detail-content"
      >
        Computers
      </div>
    </div>
    <div
      class="detail"
    >
      6. PE Ratio
      <div
        class="detail-content"
      >
        12.2
      </div>
    </div>
    <div
      class="detail"
    >
      7. Market Cap
      <div
        class="detail-content"
      >
        123345
      </div>
    </div>
  </div>
</div>
`;
